app/services/articles/feeds/find_featured_story.rb

```diff
@@ -1,9 +1,18 @@
 module Articles
   module Feeds
     module FindFeaturedStory
-      def self.call(stories)
+      # @param stories [ActiveRecord::Relation, #detect] pick a
+      #        featured story from this enumerable
+      # @param must_have_main_image [Boolean] if true, the featured
+      #        story must have a main image
+      # @return [Article]
+      #
+      # @note the must_have_main_image parameter name matches PR #15240
+      def self.call(stories, must_have_main_image: true)
         featured_story =
-          if stories.is_a?(ActiveRecord::Relation)
+          if !must_have_main_image
+            stories.first
+          elsif stories.is_a?(ActiveRecord::Relation)
             stories.where.not(main_image: nil).first
           else
             stories.detect { |story| story.main_image.present? }
```

Comment:
I feel like I must be missing something here but are these conditions outlining the same thing?

It seems we're finding the first story where there is a main_image. 

`main_image.present? checks for nil and an empty string, whereas `stories.where.not(main_image: nil).first` checks for nil specifically?

Reply:
I went with a bit of a logic two-step tango.  Normally I disdain `if !something; else`. I personally find that hard to quickly think through.  I'm going to rework and I believe it will be more immediately legible.